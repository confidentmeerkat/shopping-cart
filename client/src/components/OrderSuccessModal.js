import React from 'react';
import Modal from 'react-modal';
import RaisedButton from 'material-ui/RaisedButton';
import '../styles/OrderSuccessModal.css';

const OrderSuccessModal = (props) => (
      <Modal
        className="order-success-modal"
        isOpen={props.isOpen}
        onRequestClose={props.onRequestClose}
      >
        <div className="success">
          <h1>Success!</h1>
          <img src="/img/success.gif" />
          <br/>
          <p>
            Your order has been received. The items you've ordered will be sent to your address.
          </p>
          <RaisedButton
            onClick={props.toggle}
            className="btn"
            label="OK"
            primary={true}
          />
        </div>
      </Modal>
);

export default OrderSuccessModal;