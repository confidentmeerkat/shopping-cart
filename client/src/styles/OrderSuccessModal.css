.order-success-modal {
  width: 25%;
  height: 400px;
  margin: auto;
  margin-top: 100px;
  background: white;
  border: 1px solid white;
  border-radius: 5px;
  outline: none;
}

.order-success-modal .success {
  width: 70%;
  margin: auto;
  margin-top: 10px;
  text-align: center;
  font-family: 'Roboto';
}

.order-success-modal h1 {
  color: #00BCD4;
}

.order-success-modal img {
  max-height: 100px;
  margin: 30px 0;
}

.order-success-modal .btn {
  margin-top: 20px;
}