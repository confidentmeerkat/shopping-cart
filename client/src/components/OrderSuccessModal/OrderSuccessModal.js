import React from 'react';
import Modal from 'react-modal';
import RaisedButton from 'material-ui/RaisedButton';
import '@styles/OrderSuccessModal.css';

const OrderSuccessModal = ({ isOpen, toggle }) => (
  <Modal
    className="order-success-modal"
    isOpen={isOpen}
    onRequestClose={() => toggle('orderSuccessModalOpen')}
  >
    <div className="success">
      <h1>Success!</h1>
      <img src="/img/success.gif" />
      <br/>
      <p>
        Your order has been received. The items you've ordered will be sent to your address.
      </p>
      <RaisedButton
        onClick={() => toggle('orderSuccessModalOpen')}
        className="btn"
        label="OK"
        primary={true}
      />
    </div>
  </Modal>
);

export default OrderSuccessModal;
