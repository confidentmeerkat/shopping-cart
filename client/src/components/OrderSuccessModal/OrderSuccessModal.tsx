import * as React from 'react';
import * as Modal from 'react-modal';
import RaisedButton from 'material-ui/RaisedButton';
import { target } from '../Cart/Cart';
import '@styles/OrderSuccessModal.css';

interface Props {
  isOpen: boolean;
  toggle: (targetComponent: target) => void;
}

const OrderSuccessModal: React.SFC<Props> = ({ isOpen, toggle }) => (
  <Modal
    className="order-success-modal"
    isOpen={isOpen}
    onRequestClose={() => toggle('orderSuccessModalOpen')}
  >
    <div className="success">
      <h1>Success!</h1>
      <img src="/img/success.gif" />
      <br/>
      <p>
        Your order has been received. The items you've ordered will be sent to your address.
      </p>
      <RaisedButton
        onClick={() => toggle('orderSuccessModalOpen')}
        className="btn"
        label="OK"
        primary={true}
      />
    </div>
  </Modal>
);

export default OrderSuccessModal;
