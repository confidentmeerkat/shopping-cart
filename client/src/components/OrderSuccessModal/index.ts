import OrderSuccessModal from './OrderSuccessModal';

export default OrderSuccessModal;
