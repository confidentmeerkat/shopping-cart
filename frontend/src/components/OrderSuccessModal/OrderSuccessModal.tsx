import React from 'react';
import Modal from 'react-modal';
import RaisedButton from 'material-ui/RaisedButton';
import { ModalProps } from '@typings/modal';
import '@styles/OrderSuccessModal.css';

const OrderSuccessModal = ({ isOpen, setActiveModal }: ModalProps) => (
  <Modal
    className="order-success-modal"
    isOpen={isOpen}
    onRequestClose={() => setActiveModal(null)}
  >
    <div className="success">
      <h1>Success!</h1>
      <img src="/img/success.gif" />
      <br/>
      <p>
        Your order has been received. The items you've ordered will be sent to your address.
      </p>
      <RaisedButton
        onClick={() => setActiveModal(null)}
        className="btn"
        label="OK"
        primary={true}
      />
    </div>
  </Modal>
);

export default OrderSuccessModal;
