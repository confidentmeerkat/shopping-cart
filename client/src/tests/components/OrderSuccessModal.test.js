import React from 'react';
import { shallow } from 'enzyme';
import OrderSuccessModal from '../../components/OrderSuccessModal';

describe('<OrderSuccessModal />', () => {
  it('renders OrderSuccessModal correctly', () => {
    const wrapper = shallow(<OrderSuccessModal />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.success').children()).toHaveLength(5);
    expect(wrapper.find('img').prop('src')).toEqual('/img/success.gif');
  });
});