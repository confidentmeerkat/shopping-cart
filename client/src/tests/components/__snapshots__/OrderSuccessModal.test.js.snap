// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderSuccessModal /> renders OrderSuccessModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="order-success-modal"
  closeTimeoutMS={0}
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="success"
  >
    <h1>
      Success!
    </h1>
    <img
      src="/img/success.gif"
    />
    <br />
    <p>
      Your order has been received. The items you've ordered will be sent to your address.
    </p>
    <RaisedButton
      className="btn"
      disabled={false}
      fullWidth={false}
      label="OK"
      labelPosition="after"
      onClick={[Function]}
      primary={true}
      secondary={false}
    />
  </div>
</Modal>
`;
